src/Render/Program.cs

```diff
@@ -89,13 +89,48 @@ internal static class Program {
 		}
 
 		private static string GetTitle( MarkdownDocument doc ) {
-			var inline = doc.Descendants<HeadingBlock>().Single(
-				h => h.Level == 1
-			).Inline;
-			var titleLiteral = inline.Descendants<LiteralInline>().Single();
+
+			HeadingBlock headingBlock = GetSingleTitleOrThrow( doc );
+
+			LiteralInline titleLiteral = GetUnformattedContentOrThrow( headingBlock );
+
 			return titleLiteral.Content.ToString();
 		}
 
+		private static LiteralInline GetUnformattedContentOrThrow(HeadingBlock headingBlock) {
+			var literalInlines =
+				headingBlock
+					.Inline
+					.Descendants<LiteralInline>()
+					.ToArray();
+
+			if( literalInlines.Length > 1 ) {
+				throw new ArgumentException( "The level-1 heading must be unformatted." );
+			}
+
+			var titleLiteral = literalInlines.Single();
+			return titleLiteral;
+		}
+
+		private static HeadingBlock GetSingleTitleOrThrow( MarkdownDocument doc ) {
+			var titles =
+				doc
+					.Descendants<HeadingBlock>()
+					.Where( h => h.Level == 1 )
+					.ToArray();
+
+			if( titles.Length == 0 ) {

```

Comment:
I moved away from `SingleOrDefault` because it throws if there are multiple entries.